pyqtgraph/parametertree/parameterTypes/basetypes.py

```diff
@@ -319,24 +319,37 @@ def __init__(self, param, depth):
         self.optsChanged(self.param, self.param.opts)
 
     def updateDepth(self, depth):
-        ## Change item's appearance based on its depth in the tree
-        ## This allows highest-level groups to be displayed more prominently.
-        if depth == 0:
-            for c in [0,1]:
-                self.setBackground(c, QtGui.QBrush(QtGui.QColor(100,100,100)))
-                self.setForeground(c, QtGui.QBrush(QtGui.QColor(220,220,255)))
-                font = self.font(c)
-                font.setBold(True)
+        """
+        Change item's appearance based on its depth in the tree
+        This allows highest-level groups to be displayed more prominently.
+        """
+        app = QtWidgets.QApplication.instance()
+        palette = app.palette()
+        textColor = palette.text().color()
+        backgroundColor = palette.button().color().darker(110)
+        altBackgroundColor = palette.dark().color()
```

Comment:
I like that this pulls colors from the palette now!

Is it necessary to mix and match the color roles here?
There are already `base()` and `alternateBase()` colors, how far can we get with those?
Mixing in `button()` as an additional background color might work, but we don't really know how that might relate to the base colors depending on platform or theme.

Reply:
This is exactly what should be used according to the documentation and it does work as described in light mode. 
![image](https://user-images.githubusercontent.com/5187375/142492659-8e691497-9f73-4266-85f8-75647b60f4e6.png)

However, in darkMode I tried using `base` and `alternateBase` but the color for `alternateBase` is _much_ brighter than it should be. 

![image](https://user-images.githubusercontent.com/5187375/142491975-7fb94fbc-7bb0-4e91-8fe4-b970507f4896.png)

There are other quirks when using the palette. You can see how initially the `altBackgroundColor` is set to `palette.dark()`, this works in light mode, but the color of `dark` doesn't change when in `darkMode`, so you end up with something like `#888`, which is fairly bright in `darkMode`.

Reply:
Which bindings/versions of qt are you using for this testing @Wubbzi ? 

I'm thinking that `AlternateBase` is what we want, but the fact that it's not specified well by Qt itself kinda makes me want to say "not our problem"...

Reply:
I switched to PySide2 5.15.2 when for testing QDarkStyle since they don't seem to support v6 yet. PySide6.2.1 looks the same. 

If everyone is fine with `base` and `alternateBase` across the board, this should be much less complicated. 😄